src/daft-micropartition/src/ops/cast_to_schema.rs

```diff
@@ -26,12 +27,24 @@ impl MicroPartition {
         let guard = self.state.lock().unwrap();
         match guard.deref() {
             // Replace schema if Unloaded, which should be applied when data is lazily loaded
-            TableState::Unloaded(scan_task) => Ok(MicroPartition::new_unloaded(
-                schema.clone(),
-                scan_task.clone(),
-                self.metadata.clone(),
-                pruned_statistics.expect("Unloaded MicroPartition should have statistics"),
-            )),
+            TableState::Unloaded(scan_task) => {
```

Comment:
Note that in the Unloaded case, the fill_map is only applied to stats and does not correctly propagate to the data. This is because there isn't a way to correctly parametrize an unloaded MicroPartition with a fillmap.

Because of this, it's a little odd to call `cast_to_schema_with_fillmap` on an unloaded MicroPartition, and after this PR we no longer do that on the native downloader codepaths.